### PR TITLE
Migrate Food101 prototype dataset

### PR DESCRIPTION
**Pull request by NicolasHug**

### CODE REVIEW

File: torchvision/prototype/datasets/_builtin/food101.py [hunk 1]
```diff
@@ -83,9 +95,15 @@ def _make_datapipe(
...
+        dp = resources[0].load(self._root)
         dp = Filter(dp, path_comparator("name", "classes.txt"))
         dp = LineReader(dp, decode=True, return_path=False)
         return list(dp)
+
+    def __len__(self) -> int:
+        return {
+            "train": 75_750,
+            "test": 25_250,
+        }[self._split]
```
Comment (pmeier) [inline torchvision/prototype/datasets/_builtin/food101.py:109]: Nit: do we always want to go with dicts here? For two values, I went for a simple `if` / `else`

```suggestion
        return 75_750 if self._split == "train" else 25_250
```

No strong opinion though.

**Review by pmeier — APPROVED** [review]
One nit, otherwise LGTM. Thanks Nicolas!

**Comment by pmeier** [comment]
cc @yassineAlouini This is the PR that moves the Food101 dataset to the new style.

**Comment by yassineAlouini** [comment]
> cc @yassineAlouini This is the PR that moves the Food101 dataset to the new style.

Thanks @pmeier for the link. This looks neat, well done to the team.

**Pull request merged**